warehouse/integrations/vulnerabilities/utils.py

```diff
@@ -0,0 +1,200 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import List
+
+from sqlalchemy import func
+from sqlalchemy.orm.exc import NoResultFound
+
+from warehouse.integrations import verifier
+from warehouse.metrics import IMetricsService
+from warehouse.packaging.models import Project, Release, Vulnerability
+
+
+class InvalidVulnerabilityReport(Exception):
+    def __init__(self, message, reason):
+        self.reason = reason
+        super().__init__(message)
+
+
+class VulnerabilityReport:
+    def __init__(
+        self,
+        project: str,
+        versions: List[str],
+        vulnerability_id: str,
+        advisory_link: str,
+        aliases: List[str],
+    ):
```

Comment:
I think we can probably condense the `Vulnerability` database model and this class into a single class. The class can also live at `warehouse/integrations/vulnerabilities/models.py` instead, and be imported into `warehouse.packaging.models`.

Reply:
As in, do the model representation, report validating, and DB interaction in a single class?